### ISSUE TITLE
Move to a new hosting provider

### ISSUE THREAD

**Issue by mulholio** [eat-well/nutrimprove#3]
# Hosting and project structure

#### TL;DR: I propose we switch to a serverless monorepo and host it on Now.

Currently, EatWell is setup using an AWS hosted MariaDB instance and deployed by SSHing into a private server. I propose that we upgrade this setup and use something simpler and more automatable.

The current setup involves manual deploys. As well as being slow, this means we miss out on the advantages of using a modern service (e.g. Heroku, Now) such as automatic staging, GitHub integrations/automation, rollbacks and a UI. Using Now/Heroku will also help us work better as a team as these services allow team logins where we can all view the project’s status. Presumably, you are also paying something for the current server and there is the opportunity to run serverless functions for free whilst we are in development phase.

Relatedly, I think we can use this as an opportunity to improve the code structure and setup. More specifically, we can improve the development setup by using a local DB and better mirror the production environment on your local machine. If we move to a serverless model on Now, we can use the now dev command to quickly see what a production deployment will look like.

I propose we move to Now but I’d also be open to using something like Heroku if you think sticking to the serverful model is a good idea. I’ll put forward my argument for Now but I can write up a case for Heroku or other if we run into any serious objections for using Now.

## The Case for Serverless Deployment with Now

I’ve used Now in the past and had very positive experiences with it. It is built by Zeit, the team behind Next.js so there is a good integration between the two and good documentation.

Hosting on Now would mean using a serverless model. At such an early stage in the project, this would be a minimal change to the code. Serverless in general (beyond the Now platform) offers the following benefits:

Scalable performance and pricing: rather than having one constant server, we can have functions that scale easily with demand so that users get good service and we don’t pay for code that isn’t running.
Low configuation
More specifically to the Now platform, we would also get the following benefits:

Abstraction of most of the DevOps work, almost everything being handled by a simple JSON config file. All we have to do is put the API code in one folder, Next.js app in another and we are good to go.
Simpler Next.js setup. Ditching app.js and using the pre-configured serverless Next.js build reduces configuration overhead for the Next.js end of things.
GitHub integration allowing automatic deployment and preview deploys for pull requests. This would mean easy access and deployment for all team members (with the appropriate permissions)
A very good CDN
Fast deployment, only updating the code that has changed
A friendlier UI
Domain configuration and free SSL
A free tier
This guide from Zeit gives an example of the kind of setup we could have: https://zeit.co/guides/deploying-next-and-mysql-with-now

All in all, I think this change (or something like it) is moderately high priority and will allow quicker, safer, higher-performance development moving forwards.

----

### Edit: Decision Approved. Keeping here for reference purposes.

----

## Todos
- [x] Configure Now organisation
- [ ] Split API logic into /api folder lambdas
  - [x] Configure/write `db.js` to manage DB connections
  - [x] `food/id/:id`
  - [x] `food/name/:name`
  - [x] `foods`
  - [x] `recommendations`
- [x] Add `next.config.js`
- [x] Add `now.json`
- [x] Remove app.js
- [x] Document usage, including `now dev` command
  - .env can only be used in lambdas (for now)
- [x] Add now secrets
- [x] Merge and test GitHub deploy

**Comment by WhiT3Kr0w** [comment]
Done. Used NextJS api routes.
